lib/charms/mongodb/v0/set_status.py

```diff
@@ -54,19 +54,51 @@ def set_and_share_status(self, status: StatusBase):
     def set_app_status(self):
         """TODO Future Feature/Epic: parse statuses and set a status for the entire app."""
 
-    def are_all_units_ready_for_upgrade(self) -> bool:
+    def is_current_unit_ready(self, ignore_unhealthy_upgrade: bool = False) -> bool:
+        """Returns True if the current unit status shows that the unit is ready.
+
+        Note: we allow the use of ignore_unhealthy_upgrade, to avoid infinite loops due to this
+        function returning False g and preventing the status from being reset.
+        """
+        if isinstance(self.charm.unit.status, ActiveStatus):
+            return True
+
+        if ignore_unhealthy_upgrade and self.charm.unit.status == Config.Status.UNHEALTHY_UPGRADE:
+            return True
+
+        return self.is_status_related_to_mismatched_revision(
+            str(type(self.charm.unit.status)).lower()
```

Comment:
This will return something along the lines of `"<class 'ops.model.ActiveStatus'>"`, may be safer to get the class name directly:
```suggestion
            type(self.charm.unit.status).__name__.lower()
```